In [80]:
# import libraries will be used
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as ex
from plotly.subplots import make_subplots
import plotly.graph_objects as go
import folium
from folium.plugins import HeatMap
from geopy.geocoders import Nominatim
import json

## Import Data from saved file 

In [81]:
df = pd.read_csv('Finale Merged Data.csv')

## Some Information about data

In [82]:
df.columns

Index(['Patient_Name', 'Patient_Phone', 'Patient_Age', 'Patient_Address',
       'Doctor_Phone_x', 'Patient_Gender', 'Follower_Phone',
       'Doctor_National_ID', 'Age_Normalization_Patient', 'Patient_Age_Stage',
       'Result_ID', 'Result', 'Result_Date', 'Medical_Result_Year',
       'Medical_Result_Month', 'Doctor_ID', 'Diabetic', 'Hypertension',
       'Cholesterol', 'survey_id', 'q_one', 'q_two', 'q_three', 'q_four',
       'q_five', 'q_six', 'q_seven', 'q_eight', 'q_nine', 'q_ten',
       'survey_date', 'Survay_Result_Month', 'Survay_Result_Year',
       'Patient_National_ID', 'Doctor_Name', 'Doctor_Phone_y',
       'Doctor_Address', 'Doctor_Age', 'Doctor_Age_Normalization',
       'Doctor_Age_Stage'],
      dtype='object')

In [83]:
df.head()

Patient_Name  Patient_Phone  Patient_Age Patient_Address  \
0             Ali Bakr     1025486358           30         Mallawi   
1  Abdelrahman Mohamed     1045863251           20           Cairo   
2          Karem Assem     1235896541           22           Minia   
3         Hazem Khaled     1028963547           26         Mansora   
4        Mustafa Yahia     1036587460           28          Assuit   

   Doctor_Phone_x  Patient_Gender  Follower_Phone  Doctor_National_ID  \
0      1258963251               0      1158263541      20152563201253   
1      1258963251               0      1024589637      20301548520137   
2      1258963251               0      1253658477      20315059632104   
3      1258963251               0      1258963457      20102365215422   
4      1258963251               0      1236584754       1025485236541   

   Age_Normalization_Patient Patient_Age_Stage  ...  survey_date  \
0                       0.55             Adult  ...   2022-04-20   
1                       1.24             Adult  ...   2016-07-24   
2                       1.10             Adult  ...   2022-06-22   
3                       0.82             Adult  ...   2022-02-22   
4                       0.68             Adult  ...   2018-12-12   

   Survay_Result_Month Survay_Result_Year  Patient_National_ID    Doctor_Name  \
0                    4               2022       20320154852369       Zain Ali   
1                    7               2016       30201548596352      Karma Ali   
2                    6               2022       30245849635712    Muhra Malek   
3                    2               2022       42589635715284   Hamza Khaled   
4                   12               2018       20301548967415  Ahmed Mahmoud   

   Doctor_Phone_y  Doctor_Address  Doctor_Age  Doctor_Age_Normalization  \
0      1024586351         Mallawi          28                      1.19   
1      1236548520        Mansoura          30                      1.01   
2      1152480236           Cairo          31                      0.92   
3      1025879631          Assuit          45                      0.33   
4      1002542023           Luxor          34                      0.66   

   Doctor_Age_Stage  
0             Adult  
1             Adult  
2             Adult  
3             Adult  
4             Adult  

[5 rows x 40 columns]

In [84]:
df.shape

(7, 40)

In [85]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7 entries, 0 to 6
Data columns (total 40 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Patient_Name               7 non-null      object 
 1   Patient_Phone              7 non-null      int64  
 2   Patient_Age                7 non-null      int64  
 3   Patient_Address            7 non-null      object 
 4   Doctor_Phone_x             7 non-null      int64  
 5   Patient_Gender             7 non-null      int64  
 6   Follower_Phone             7 non-null      int64  
 7   Doctor_National_ID         7 non-null      int64  
 8   Age_Normalization_Patient  7 non-null      float64
 9   Patient_Age_Stage          7 non-null      object 
 10  Result_ID                  7 non-null      int64  
 11  Result                     7 non-null      int64  
 12  Result_Date                7 non-null      object 
 13  Medical_Result_Year        7 non-null      int64  
 14

In [86]:
df.describe()

Patient_Phone  Patient_Age  Doctor_Phone_x  Patient_Gender  \
count   7.000000e+00     7.000000    7.000000e+00        7.000000   
mean    1.093904e+09    24.428571    1.258963e+09        0.285714   
std     1.052159e+08     3.644957    0.000000e+00        0.487950   
min     1.025486e+09    20.000000    1.258963e+09        0.000000   
25%     1.027422e+09    22.000000    1.258963e+09        0.000000   
50%     1.036587e+09    23.000000    1.258963e+09        0.000000   
75%     1.140880e+09    27.000000    1.258963e+09        0.500000   
max     1.258651e+09    30.000000    1.258963e+09        1.000000   

       Follower_Phone  Doctor_National_ID  Age_Normalization_Patient  \
count    7.000000e+00        7.000000e+00                   7.000000   
mean     1.205935e+09        1.660053e+13                   0.931429   
std      8.751395e+07        7.222384e+12                   0.252615   
min      1.024590e+09        1.025485e+12                   0.550000   
25%      1.197424e+09        1.715270e+13                   0.750000   
50%      1.253658e+09        2.010365e+13                   1.030000   
75%      1.254743e+09        2.022706e+13                   1.100000   
max      1.258963e+09        2.031506e+13                   1.240000   

         Result_ID    Result  Medical_Result_Year  ...   q_seven   q_eight  \
count     7.000000  7.000000             7.000000  ...  7.000000  7.000000   
mean   5423.000000  0.571429          2021.285714  ...  0.571429  0.428571   
std       2.160247  0.534522             1.112697  ...  0.534522  0.534522   
min    5420.000000  0.000000          2020.000000  ...  0.000000  0.000000   
25%    5421.500000  0.000000          2020.500000  ...  0.000000  0.000000   
50%    5423.000000  1.000000          2021.000000  ...  1.000000  0.000000   
75%    5424.500000  1.000000          2022.000000  ...  1.000000  1.000000   
max    5426.000000  1.000000          2023.000000  ...  1.000000  1.000000   

         q_nine     q_ten  Survay_Result_Month  Survay_Result_Year  \
count  7.000000  7.000000             7.000000            7.000000   
mean   0.571429  0.285714             6.428571         2020.285714   
std    0.534522  0.487950             3.866831            2.627691   
min    0.000000  0.000000             2.000000         2016.000000   
25%    0.000000  0.000000             3.500000         2018.500000   
50%    1.000000  0.000000             6.000000         2022.000000   
75%    1.000000  0.500000             9.000000         2022.000000   
max    1.000000  1.000000            12.000000         2023.000000   

       Patient_National_ID  Doctor_Phone_y  Doctor_Age  \
count         7.000000e+00    7.000000e+00    7.000000   
mean          2.992658e+13    9.395913e+08   36.142857   
std           8.922419e+12    3.748380e+08    8.234654   
min           2.030155e+13    1.125480e+08   28.000000   
25%           2.283609e+13    1.012548e+09   30.500000   
50%           3.020155e+13    1.024586e+09   34.000000   
75%           3.536058e+13    1.089180e+09   40.000000   
max           4.258964e+13    1.236549e+09   50.000000   

       Doctor_Age_Normalization  
count                  7.000000  
mean                   0.778571  
std                    0.289161  
min                    0.330000  
25%                    0.615000  
50%                    0.770000  
75%                    0.965000  
max                    1.190000  

[8 rows x 29 columns]

### Remove Columns not needed in analysis 

In [87]:
df.shape[1]

40

In [88]:
df = df.drop(['Patient_Name', 'Patient_Phone','Doctor_Phone_x', 
                'Result_ID' , 'Follower_Phone', 'survey_id', 'Doctor_Phone_y'  ], axis = 1)

In [89]:
df.columns

Index(['Patient_Age', 'Patient_Address', 'Patient_Gender',
       'Doctor_National_ID', 'Age_Normalization_Patient', 'Patient_Age_Stage',
       'Result', 'Result_Date', 'Medical_Result_Year', 'Medical_Result_Month',
       'Doctor_ID', 'Diabetic', 'Hypertension', 'Cholesterol', 'q_one',
       'q_two', 'q_three', 'q_four', 'q_five', 'q_six', 'q_seven', 'q_eight',
       'q_nine', 'q_ten', 'survey_date', 'Survay_Result_Month',
       'Survay_Result_Year', 'Patient_National_ID', 'Doctor_Name',
       'Doctor_Address', 'Doctor_Age', 'Doctor_Age_Normalization',
       'Doctor_Age_Stage'],
      dtype='object')

In [90]:
df.shape[1]

33

In [91]:
df.head(7)

Patient_Age Patient_Address  Patient_Gender  Doctor_National_ID  \
0           30         Mallawi               0      20152563201253   
1           20           Cairo               0      20301548520137   
2           22           Minia               0      20315059632104   
3           26         Mansora               0      20102365215422   
4           28          Assuit               0       1025485236541   
5           22           Sohag               1      20103654852012   
6           23            Giza               1      14203025698410   

   Age_Normalization_Patient Patient_Age_Stage  Result Result_Date  \
0                       0.55             Adult       0  2022-09-10   
1                       1.24             Adult       1  2020-12-04   
2                       1.10             Adult       0  2021-04-05   
3                       0.82             Adult       1  2022-06-30   
4                       0.68             Adult       1  2023-05-01   
5                       1.10             Adult       0  2021-08-14   
6                       1.03             Adult       1  2020-01-16   

   Medical_Result_Year  Medical_Result_Month  ...  q_ten  survey_date  \
0                 2022                     9  ...      1   2022-04-20   
1                 2020                    12  ...      0   2016-07-24   
2                 2021                     4  ...      1   2022-06-22   
3                 2022                     6  ...      0   2022-02-22   
4                 2023                     5  ...      0   2018-12-12   
5                 2021                     8  ...      0   2023-03-08   
6                 2020                     1  ...      0   2019-11-11   

   Survay_Result_Month  Survay_Result_Year  Patient_National_ID  \
0                    4                2022       20320154852369   
1                    7                2016       30201548596352   
2                    6                2022       30245849635712   
3                    2                2022       42589635715284   
4                   12                2018       20301548967415   
5                    3                2023       25352015984520   
6                   11                2019       40475302156985   

     Doctor_Name  Doctor_Address  Doctor_Age  Doctor_Age_Normalization  \
0       Zain Ali         Mallawi          28                      1.19   
1      Karma Ali        Mansoura          30                      1.01   
2    Muhra Malek           Cairo          31                      0.92   
3   Hamza Khaled          Assuit          45                      0.33   
4  Ahmed Mahmoud           Luxor          34                      0.66   
5   Nour Mohamed           Dahab          50                      0.77   
6     Maleka Ali      Alexandria          35                      0.57   

   Doctor_Age_Stage  
0             Adult  
1             Adult  
2             Adult  
3             Adult  
4             Adult  
5             Adult  
6             Adult  

[7 rows x 33 columns]

Result            1.00
q_seven           0.42
q_six             0.17
q_eight           0.17
q_two             0.09
q_five            0.09
Patient_Gender   -0.09
q_one            -0.09
q_four           -0.09
Cholesterol      -0.17
q_nine           -0.17
q_three          -0.55
Hypertension     -0.73
q_ten            -0.73
Diabetic         -0.75
Name: Result, dtype: float64


- Positive coefficients indicate a positive correlation
- Negative coefficients indicate a negative correlation.
- The magnitude of the coefficient indicates the strength of the correlation.

## Injured Male & female


In [92]:
# Convert Result, Patient_Gender to int datatypes
df['Result'] = df['Result'].astype('int')
df['Patient_Gender'] = df['Patient_Gender'].astype('int')

In [93]:
df['Result'] = df['Result'].astype(str)
df['Patient_Gender'] = df['Patient_Gender'] .astype(str)
# Replace the values in the 'resultlabel' column
df['Result'] = df['Result'].replace({'0': 'Not Injured', '1': 'Injured'})
df['Patient_Gender'] = df['Patient_Gender'].replace({'0': 'Male', '1': 'Female'})

In [94]:
# Group the DataFrame by 'resultlabel' and 'Gender' columns and count the occurences
grouped_df_resultlabel_gender = df.groupby(['Result', 'Patient_Gender']).size().reset_index(name='counts')

In [95]:
# Create a stacked bar chart of the counts grouped by 'resultlabel' and 'Gender' using Plotly Express
fig = ex.bar(grouped_df_resultlabel_gender, x = 'Result', y = 'counts', color = 'Patient_Gender',
             barmode = 'stack', color_discrete_sequence=['darkviolet', 'plum'])

# Add title and axis labels
fig.update_layout(title = 'Counts of Injured Result by Gender')
fig.update_xaxes(title = 'Injured Result')
fig.update_yaxes(title = 'Count of Male & Female')

fig.update_layout(width=600, height=400)
# Display the chart using Graph Objects
fig.show()

**Observation**:
- Males have more infections than females.

## Max months of injuries

In [96]:
# Rename Survay_Result_Year 
df = df.rename(columns = {'Survay_Result_Year':'Survay_Year'})

In [97]:
df.Medical_Result_Year = df.Medical_Result_Year.astype(int)
df.Survay_Year = df.Survay_Year.astype(int)

In [98]:
# replace 1 with "Injured" and 0 with "Not_Injured" in the "Result" column
df['Result_Inj_not'] = df['Result'].replace({1: 'Injured', 0: 'Not_Injured'})



In [99]:
# temporary dataframe containing chronic diseases as: Hypertension, Cholesterol, Diabetic and the result of detection
temp = df[['Hypertension', 'Cholesterol', 'Diabetic','Result_Inj_not']]
temp

Hypertension  Cholesterol  Diabetic Result_Inj_not
0         False         True      True    Not Injured
1         False        False     False        Injured
2          True         True      True    Not Injured
3         False         True     False        Injured
4         False        False      True        Injured
5          True        False      True    Not Injured
6         False         True     False        Injured

In [100]:
disease_counts = temp.groupby('Result_Inj_not')['Hypertension', 'Diabetic', 'Cholesterol'].sum()
disease_counts

C:\Users\sohil\AppData\Local\Temp\ipykernel_25016\3444983822.py:1: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



Hypertension  Diabetic  Cholesterol
Result_Inj_not                                     
Injured                    0         1            2
Not Injured                2         3            2

In [101]:
index = pd.MultiIndex.from_product([['Injured', 'Not Injured'], ['Result_Inj_not']])


# create heatmap chart using plotly
fig = go.Figure(data=go.Heatmap(z = disease_counts.values,
                                 x = disease_counts.columns,
                                 y = disease_counts.index.get_level_values(0)))

# update chart layout and axis labels
fig.update_layout(title='Patient Medical Conditions by Injury Status',
                  xaxis_title='Chronic Diseases',
                  yaxis_title='Injury Status')

fig.update_layout(width=800, height=400)
# show the chart
fig.show()

## Injuries Patients by Age Stage

In [102]:
# Group data of 'Patient_Age_Stage' and 'Result' with number of Injured and not injured patients.
cnt = df.groupby(['Patient_Age_Stage', 'Result_Inj_not']).size().reset_index(name='counts')
# Pivot data and make Injured and not injured number is columns
cnt = cnt.pivot(index = 'Patient_Age_Stage', columns = 'Result_Inj_not', values = 'counts').reset_index()
# Set index is 'Patient_Age_Stage'
cnt = pd.DataFrame(cnt).set_index('Patient_Age_Stage')
cnt

Result_Inj_not     Injured  Not Injured
Patient_Age_Stage                      
Adult                    4            3

In [103]:


# create MultiIndex from columns 'Not Injured' and 'Injured'
cnt.columns = pd.MultiIndex.from_product([cnt.columns, ['Count']])

# create heatmap chart using plotly
fig = go.Figure(data=go.Heatmap(z = cnt.values,
                                 x = cnt.columns.levels[0],
                                 y = cnt.index))

# update chart layout and axis labels
fig.update_layout(title='Injuries Patients by Age Stage',
                  xaxis_title='Injury Status',
                  yaxis_title='Age Stage')

# show the chart
fig.show()

## Number of Injuries within Age Status

In [104]:
# Group data of 'Patient_Age_Stage' and 'Result' with number of Injured and not injured patients.
cnt = df.groupby(['Patient_Age_Stage', 'Result']).size().reset_index(name='counts')
# Pivot data and make Injured and not injured number is columns
cnt = cnt.pivot(index='Patient_Age_Stage', columns='Result', values= 'counts').reset_index()
# Create a figure 
fig = go.Figure()
fig.add_trace(go.Bar(y = cnt.Injured, x = cnt.Patient_Age_Stage,
              name = 'Injured', marker_color = 'slateblue'))
fig.add_trace(go.Bar(y = cnt['Not Injured'], x = cnt.Patient_Age_Stage,
              name = 'Not Injured', marker_color = 'darkslateblue'))
fig.update_layout(title='Number of Injuries within Age Status')

fig.update_layout(width = 800, height = 400)
fig.show()


العلاقة بين العمر و الاصابة

In [105]:
# Make temp contian subset of data of just injured patients

In [106]:
injured = df[df['Result'] == 'Injured']
#injured = df[['Patient_Age_Stage' ]]
injured

Patient_Age Patient_Address Patient_Gender  Doctor_National_ID  \
1           20           Cairo           Male      20301548520137   
3           26         Mansora           Male      20102365215422   
4           28          Assuit           Male       1025485236541   
6           23            Giza         Female      14203025698410   

   Age_Normalization_Patient Patient_Age_Stage   Result Result_Date  \
1                       1.24             Adult  Injured  2020-12-04   
3                       0.82             Adult  Injured  2022-06-30   
4                       0.68             Adult  Injured  2023-05-01   
6                       1.03             Adult  Injured  2020-01-16   

   Medical_Result_Year  Medical_Result_Month  ...  survey_date  \
1                 2020                    12  ...   2016-07-24   
3                 2022                     6  ...   2022-02-22   
4                 2023                     5  ...   2018-12-12   
6                 2020                     1  ...   2019-11-11   

   Survay_Result_Month  Survay_Year  Patient_National_ID    Doctor_Name  \
1                    7         2016       30201548596352      Karma Ali   
3                    2         2022       42589635715284   Hamza Khaled   
4                   12         2018       20301548967415  Ahmed Mahmoud   
6                   11         2019       40475302156985     Maleka Ali   

   Doctor_Address  Doctor_Age  Doctor_Age_Normalization  Doctor_Age_Stage  \
1        Mansoura          30                      1.01             Adult   
3          Assuit          45                      0.33             Adult   
4           Luxor          34                      0.66             Adult   
6      Alexandria          35                      0.57             Adult   

   Result_Inj_not  
1         Injured  
3         Injured  
4         Injured  
6         Injured  

[4 rows x 34 columns]

In [107]:
Age_stage_value_counts = injured.Patient_Age_Stage.value_counts()
Age_stage_value_counts

Adult    4
Name: Patient_Age_Stage, dtype: int64

In [108]:
fig = ex.bar(Age_stage_value_counts, x=Age_stage_value_counts.index, y=Age_stage_value_counts.values, 
            title='Number of Injured Patient within each Age Stage')

fig.update_layout(width = 600, height = 400,  yaxis_title='Count of Injured')
fig.show()

## Governments with injuries 

In [109]:
govr_with_injured = injured.groupby('Patient_Address')['Patient_National_ID'].count()
govr_with_injured

Patient_Address
Assuit     1
Cairo      1
Giza       1
Mansora    1
Name: Patient_National_ID, dtype: int64

In [110]:
for government, location in locations.items():
    # count the number of times the government name appears in the list
    count = government_list.count(government)
    # add a circle for each occurrence of the government name
    for i in range(count):
        folium.Circle(location=location,
                      radius=50000,
                      color='red',
                      fill=True,
                      fill_color='red',
                      fill_opacity=0.5,
                      tooltip=f"{government} ({count})").add_to(mp)   

NameError: name 'locations' is not defined

In [ ]:
mp = folium.Map(location=[26.8206, 30.8025], zoom_start=6)
for gov in government_list:
    geolocator = Nominatim(user_agent='my-app')
    place_name = '{}, Egypt'.format(gov)
    location = geolocator.geocode(place_name)
    lat = location.latitude
    lon = location.longitude
    folium.Marker(location=[lat, lon], tooltip = gov).add_to(mp)
    folium.Circle(location=[lat, lon],
                      radius=500,
                      color='red',
                      fill=True,
                      fill_color='red',
                      fill_opacity=0.5,
                      tooltip=f"{government} ({count})").add_to(mp)   
mp    

In [ ]:
#  create subset from df within injured results
injured_address = df[df['Result'] == 'Injured']
injured_address = injured_address[['Result', 'Patient_Address']]
# group by address and number of injure
injured_count = injured_address.groupby('Patient_Address')['Result'].count().sort_values()
# create a horizontal bar chart
fig = go.Figure()
fig.add_trace(go.Bar(
    x=injured_count.values,
    y=injured_count.index,
    orientation = 'h'
))

# set the chart title and axis labels
fig.update_layout(
    title='Number of Injured Patients by Address',
    xaxis_title='Count',
    yaxis_title='Patient Address'
)
# set the y-axis ticks to integers
fig.update_xaxes(tickmode = 'linear', dtick = 1, title = 'Number of Injured Patients')
# show the chart
fig.show()

### Relation between Diabetic, Cholesterol, Hypertension and Injuries by Pneumonia

In [ ]:
df.Diabetic.astype(int)
df['Result_0_1'] = df.Result.map({'Injured': 1, 'Not Injured': 0})
df['Result_0_1']

0    0
1    1
2    0
3    1
4    1
5    0
6    1
Name: Result_0_1, dtype: int64

In [ ]:
corr = df['Diabetic'].corr(df['Result_0_1'])

print("Correlation coefficient between Diabetic and Result columns:", corr)

Correlation coefficient between Diabetic and Result columns: -0.75


In [ ]:
corr = df['Hypertension'].corr(df['Result_0_1'])

print("Correlation coefficient between Hypertension and Result columns:", corr)

Correlation coefficient between Hypertension and Result columns: -0.7302967433402214


In [ ]:
corr = df['Cholesterol'].corr(df['Result_0_1'])

print("Correlation coefficient between Cholesterol and Result columns:", corr)

Correlation coefficient between Cholesterol and Result columns: -0.16666666666666669


#### **Abstract**
- columns indicates a strong negative linear relationship between the two variables. This means that as the value of one variable (e.g., 'Diabetic') increases, the value of the other variable (e.g., 'Result') tends to decrease, and vice versa.
- A correlation coefficient of -0.7302967433402214 between the 'Cholesterol' and 'Result' columns indicates a strong negative linear relationship between the two variables. This means that as the value of one variable (e.g., 'Diabetic') increases, the value of the other variable (e.g., 'Result') tends to decrease, and vice versa.
- A correlation coefficient of -0.16666666666666669 between the 'Diabetic' and 'Result' columns indicates a weak negative linear relationship between the two variables. This means that as the value of one variable (e.g., 'Diabetic') increases, the value of the other variable (e.g., 'Result') tends to decrease slightly, and vice versa.

## Number of injuries within each Year

In [ ]:
year = injured.Medical_Result_Year.value_counts().to_frame().rename(columns ={'Medical_Result_Year':'counts'})
year

counts
2020       2
2022       1
2023       1

In [ ]:
# Creat a car chart using injured people in each year
fig = ex.bar(year, x = year.counts.index,
                    y = year.counts.values, 
                    title = 'Count of Injuries by Pneumonia', 
                    color_discrete_sequence = px.colors.qualitative.Pastel)

fig.update_layout(xaxis = {'type': 'category', 'tickmode': 'array', 'tickvals': year.counts.index},
                        yaxis={'tickmode': 'array', 'tickvals': list(range(0, year.counts.values.max() + 1))})
# Add titles for each axes                        
fig.update_yaxes(title = 'Number of Injured Patients')
fig.update_xaxes(title = 'Year')
fig.update_layout(width=600, height=400)
fig.show()

In [ ]:
df.Medical_Result_Month = df.Medical_Result_Month.astype(int)

In [ ]:
year_month = df.groupby('Medical_Result_Year')['Medical_Result_Month'].value_counts().to_frame().rename(columns={'Medical_Result_Month': 'Count'})
year_month = year_month.reset_index().set_index('Medical_Result_Year')
year_month

Medical_Result_Month  Count
Medical_Result_Year                             
2020                                    1      1
2020                                   12      1
2021                                    4      1
2021                                    8      1
2022                                    6      1
2022                                    9      1
2023                                    5      1

In [ ]:
year_month.index

Int64Index([2020, 2020, 2021, 2021, 2022, 2022, 2023], dtype='int64', name='Medical_Result_Year')

In [ ]:
year_month.dtypes

Medical_Result_Month    int64
Count                   int64
dtype: object

In [ ]:
year_month['Medical_Result_Month']


Medical_Result_Year
2020     1
2020    12
2021     4
2021     8
2022     6
2022     9
2023     5
Name: Medical_Result_Month, dtype: int64

In [ ]:
year_month[ year_month.index == 2022]

Medical_Result_Month  Count
Medical_Result_Year                             
2022                                    6      1
2022                                    9      1

In [ ]:
# List unique years in year_month
years = year_month.index.get_level_values('Medical_Result_Year').unique().tolist()
# make subplots figure
fig = make_subplots(rows=len(years), cols=1, subplot_titles=years)

for i, year in enumerate(years):
    data = year_month.loc[year].reset_index()
    fig.add_trace(go.Bar(x=year_month['Medical_Result_Month'], y=year_month['Count'], name=str(year)), row=i+1, col=1)
    #
    fig.update_xaxes(title_text='Medical_Result_Month', row=i+1, col=1)
    # 
    fig.update_yaxes(title_text='Count', row=i+1, col=1)
'''
    fig.update_xaxes(title_text='Medical_Result_Month', row= i+1, col=1, tickmode='linear', dtick=1)
    fig.update_yaxes(title_text='Count', row=i+1, col=1, tickmode='linear', dtick=1)'''

fig.update_layout(title='Histogram of Injuries by Pneumonia by Year', height=500*len(years))
fig.show()

## Number of people who said that they have the mentioned disease

In [ ]:
questuion_df = df[['q_one', 'q_two', 'q_three', 'q_four', 'q_five', 'q_six', 'q_seven', 'q_eight', 'q_nine', 'q_ten']]
rotated_df = questuion_df.transpose()
# create a new column 'total_sum' that contains the sum of all integer columns
rotated_df['total_sum'] = rotated_df.sum(axis=1)
rotated_df

0  1  2  3  4  5  6  total_sum
q_one    1  0  0  1  0  0  0          2
q_two    0  1  1  1  1  1  0          5
q_three  1  0  1  1  0  1  1          5
q_four   1  1  0  0  0  0  0          2
q_five   1  1  0  1  0  1  1          5
q_six    0  1  1  0  1  0  0          3
q_seven  0  1  0  0  1  1  1          4
q_eight  0  0  0  0  1  1  1          3
q_nine   1  0  1  1  1  0  0          4
q_ten    1  0  1  0  0  0  0          2

In [ ]:
# pd.pivot_table(df, index='Category', values='Value', aggfunc='count')

In [ ]:
temp = rotated_df.total_sum.to_frame().reset_index()
temp

index  total_sum
0    q_one          2
1    q_two          5
2  q_three          5
3   q_four          2
4   q_five          5
5    q_six          3
6  q_seven          4
7  q_eight          3
8   q_nine          4
9    q_ten          2

In [ ]:
temp.index

RangeIndex(start=0, stop=10, step=1)

In [ ]:
fig = ex.bar(temp, x = temp['index'], y = temp['total_sum'], title = 'Number of people who said that they have the mentioned disease')
fig.update_layout(width=700, height=400)
fig.show()

In [ ]:
# create heatmap chart using plotly
fig = go.Figure(data=go.Heatmap(z=rotated_df.values,
                                 x=rotated_df.columns,
                                 y=rotated_df.index))

# update chart layout and axis labels
fig.update_layout(title='Responses to 10 Questions',
                  xaxis_title='Question',
                  yaxis_title='Respondent')

# show the chart
fig.show()

In [ ]:
df.columns


Index(['Patient_Age', 'Patient_Address', 'Patient_Gender',
       'Doctor_National_ID', 'Age_Normalization_Patient', 'Patient_Age_Stage',
       'Result', 'Result_Date', 'Medical_Result_Year', 'Medical_Result_Month',
       'Doctor_ID', 'Diabetic', 'Hypertension', 'Cholesterol', 'q_one',
       'q_two', 'q_three', 'q_four', 'q_five', 'q_six', 'q_seven', 'q_eight',
       'q_nine', 'q_ten', 'survey_date', 'Survay_Result_Month', 'Survay_Year',
       'Patient_National_ID', 'Doctor_Name', 'Doctor_Address', 'Doctor_Age',
       'Doctor_Age_Normalization', 'Doctor_Age_Stage', 'Result_Inj_not',
       'Result_0_1'],
      dtype='object')

# Doctor

In [ ]:
df.Doctor_Age_Stage

0    Adult
1    Adult
2    Adult
3    Adult
4    Adult
5    Adult
6    Adult
Name: Doctor_Age_Stage, dtype: object

In [ ]:
df.Doctor_Age

0    28
1    30
2    31
3    45
4    34
5    50
6    35
Name: Doctor_Age, dtype: int64

## Doctor Age Stages

In [ ]:
stage_counts = df['Doctor_Age_Stage'].value_counts()

# create a bar chart of the stage counts
fig = go.Figure(data=[go.Bar(x=stage_counts.index, y=stage_counts.values)])

# add labels and title
fig.update_layout(xaxis_title='Age Stage', yaxis_title='Count', title='Distribution of Doctor Age Stages')

# show the chart
fig.show()

In [ ]:
df.columns

Index(['Patient_Age', 'Patient_Address', 'Patient_Gender',
       'Doctor_National_ID', 'Age_Normalization_Patient', 'Patient_Age_Stage',
       'Result', 'Result_Date', 'Medical_Result_Year', 'Medical_Result_Month',
       'Doctor_ID', 'Diabetic', 'Hypertension', 'Cholesterol', 'q_one',
       'q_two', 'q_three', 'q_four', 'q_five', 'q_six', 'q_seven', 'q_eight',
       'q_nine', 'q_ten', 'survey_date', 'Survay_Result_Month', 'Survay_Year',
       'Patient_National_ID', 'Doctor_Name', 'Doctor_Address', 'Doctor_Age',
       'Doctor_Age_Normalization', 'Doctor_Age_Stage', 'Result_Inj_not',
       'Result_0_1'],
      dtype='object')

In [ ]:
doctor_counts = df.groupby('Doctor_Name')['Patient_National_ID'].count().sort_values()

In [ ]:
# create bar chart using plotly
fig = go.Figure(go.Bar(x=doctor_counts.index, y=doctor_counts.values))

# update chart layout and axis labels
fig = fig.update_layout(title='Number of Patients per Doctor',
                  xaxis_title='Doctor Name',
                  yaxis_title='Number of Patients',
                   yaxis=dict(
                      tickmode='linear',
                      dtick=1
                  ))

# show the chart
fig.show()

In [ ]:
df.Patient_Gender

0      Male
1      Male
2      Male
3      Male
4      Male
5    Female
6    Female
Name: Patient_Gender, dtype: object

In [115]:
df['Patient_Gender'] = df['Patient_Gender'].map({'Female': 1, 
                                                 'Male': 0})

In [111]:
df.Result = df.Result.map({'Injured': 1, 'Not Injured':0})
df.Result

0    0
1    1
2    0
3    1
4    1
5    0
6    1
Name: Result, dtype: int64

In [116]:
df['Patient_Gender']

0    0
1    0
2    0
3    0
4    0
5    1
6    1
Name: Patient_Gender, dtype: int64

In [117]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Load the data


# Define the target variable and predictor variables
target = 'Result'
predictors = [ 'Patient_Gender', 'Diabetic', 'Cholesterol', 'Hypertension', 'q_one', 'q_two', 'q_three', 'q_four', 'q_five', 'q_six', 'q_seven', 'q_eight', 'q_nine', 'q_ten']

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(df[predictors], df[target], test_size=0.2, random_state=42)

# Train the model
model = LinearRegression()
model.fit(X_train, y_train)

# Test the model
y_pred = model.predict(X_test)
mse = mean_squared_error(y_test, y_pred)
r_squared = model.score(X_test, y_test)
print('MSE:', mse)
print('R-squared:', r_squared)

MSE: 0.25015024038461564
R-squared: -0.0006009615384625633


In [119]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Define the target variable and predictor variables
target = 'Cholesterol'
predictors = ['Patient_Age', 'Patient_Gender', 'Diabetic', 'Cholesterol', 'Hypertension', 'q_one', 'q_two', 'q_three', 'q_four', 'q_five', 'q_six', 'q_seven', 'q_eight', 'q_nine', 'q_ten', 'Result']

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(df[predictors], df[target], test_size=0.2, random_state=42)

# Train the model
model = LinearRegression()
model.fit(X_train, y_train)

# Test the model
y_pred = model.predict(X_test)
mse = mean_squared_error(y_test, y_pred)
r_squared = model.score(X_test, y_test)
print('MSE:', mse)
print('R-squared:', r_squared)

MSE: 0.5130476170599727
R-squared: -1.0521904682398908


Result                       1.000000
Survay_Result_Month          0.506853
Patient_National_ID          0.484415
q_seven                      0.416667
Doctor_ID                    0.288675
q_eight                      0.166667
q_six                        0.166667
q_five                       0.091287
q_two                        0.091287
Age_Normalization_Patient    0.054662
Doctor_Age                  -0.021637
Medical_Result_Year         -0.040032
Patient_Age                 -0.061103
Patient_Gender              -0.091287
q_four                      -0.091287
q_one                       -0.091287
Medical_Result_Month        -0.148522
Cholesterol                 -0.166667
q_nine                      -0.166667
Doctor_National_ID          -0.464949
q_three                     -0.547723
Doctor_Age_Normalization    -0.586907
Survay_Year                 -0.728918
Hypertension                -0.730297
q_ten                       -0.730297
Diabetic                    -0.750000
Name: Result